<!DOCTYPE html>
<html lang="en">
	<head>
		
		
		<meta charset="utf-8">
		<meta http-equiv="X-UA-Compatible" content="IE=edge">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		
		<meta name="author" content="Juliette Hayer and Lokman Galal">
		<link rel="canonical" href="https://jhayer.github.io/bacterial_genomics_training/args/">
		<link rel="shortcut icon" href="../img/favicon.ico">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
		<title>Antimicrobial Resistance Genes - Bioinformatics Tutorials for Bacterial Genomics</title>
		<link href="../bootstrap/dist/css/bootstrap.min.css" rel="stylesheet">
		<link href="../css/font-awesome-5.12.2-all.min.css" rel="stylesheet">
		<link href="../css/base.css" rel="stylesheet">
		<link rel="stylesheet" href="../css/highlight-github.css">
		<script src="../bootstrap/dist/js/bootstrap.min.js"></script>
		<script src="../js/highlight.pack.js"></script>
		
		<script>
			var base_url = '..';
			var home_url = '../usage';
			var is_outer_page = false;
			
			var pageToc = [
				{title: "Antimicrobial Resistance Genes and Mutations detection", url: "#_top", children: [
						
				{title: "Prepare our computing environment", url: "#_top", children: [
				]},
				{title: "AMRFinderPlus", url: "#amrfinderplus", children: [
				]},
				{title: "CARD RGI", url: "#card-rgi", children: [
				]},
				]},
			];

		</script>
		<script src="../js/base.js"></script> 
	</head>

	<body class="inner-page">
		

		<div class="container-fluid wm-page-content">
			<a name="_top" aria-hidden="true"></a>
			






<div class="row pt-2 pb-4">
	<div class="col-sm text-center text-sm-left">
		<a href="../mlst/" class="btn btn-sm btn-outline-dark">
			<i class="fa fa-chevron-left" aria-hidden="true"></i>
			Previous
		</a>
		<a href="../mlst/" class="btn btn-sm btn-link">
			MLST
		</a>
	</div>
	

	
	<div class="col-sm text-center text-sm-right">
		<a href="../plasmids/" class="btn btn-sm btn-link">
			Plasmids
		</a>
		<a href="../plasmids/" class="btn btn-sm btn-outline-dark">
			Next
			<i class="fa fa-chevron-right" aria-hidden="true"></i>
		</a>
	</div>
	
</div>

			

			<h1 id="antimicrobial-resistance-genes-and-mutations-detection">Antimicrobial Resistance Genes and Mutations detection<a class="headerlink" href="#antimicrobial-resistance-genes-and-mutations-detection" title="Permanent link">#</a></h1>
<p>We will be using 2 different tools and their associated databases for detecting genes and mutations that could confer an antimicrobial resistance to the bacteria carrying it.
These tools are:</p>
<ul>
<li>
<p><a href="https://www.ncbi.nlm.nih.gov/pathogens/antimicrobial-resistance/AMRFinder/">AMRFinderPlus</a> from NCBI</p>
</li>
<li>
<p><a href="https://card.mcmaster.ca/home">CARD</a> <a href="https://card.mcmaster.ca/analyze/rgi">RGI</a> (Resistance Gene Identifier)</p>
</li>
</ul>
<p>In this practical you will learn to run these tools and their associated tools, to inspect the output files, and to visualise them in a graphical way.</p>
<h2 id="prepare-our-computing-environment">Prepare our computing environment<a class="headerlink" href="#prepare-our-computing-environment" title="Permanent link">#</a></h2>
<p>We will first run the appropriate <code>srun</code> command to book the computing cores (cpus) on the cluster.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>You need to ask the teacher which partition to use !</p>
</div>
<pre><code class="language-bash">srun -p SELECTED_PARTITION --cpus-per-task 2 --pty bash -i
</code></pre>
<p>You are now on a computing node, with computing 2 cpus reserved for you. That way, you can run commands interactively.</p>
<p>If you want to exit the <code>srun</code> interactive mode, press CTRL+D or type <code>exit</code></p>
<p>First, you need to identify in your results directory the fasta file corresponding to the unicycler hybrid assembly.</p>
<pre><code class="language-bash">cd results
ls -l
</code></pre>
<h2 id="amrfinderplus">AMRFinderPlus<a class="headerlink" href="#amrfinderplus" title="Permanent link">#</a></h2>
<p>First we will activate the AMRFinder container, and check the options and the different organisms datasets available for resistance mutations detection</p>
<pre><code class="language-bash">module load system/singularity/3.6.0

singularity shell path/to/amrfinder_container

amrfinder -h
amrfinder --list_organisms
</code></pre>
<p>Now that we have identified the organism option to use, we can run the command</p>
<pre><code class="language-bash">amrfinder --nucleotide K2_unicycler_scaffolds.fasta -o K2_unicycler_AMRfinder.txt --organism Klebsiella --plus --mutation_all K2_unicycler_AMRfinder_all_mut.txt
</code></pre>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>Inspect both output files and comment. Which resistance could be identified ?</p>
</div>
<p>Now that we are more used to the command line, we can prepare a generic script <code>amrfinder_script.sh</code> to run AMRFinderPlus on all our strains draft genomes, using a loop for <code>sbatch</code> the jobs. Create it in the <code>scripts</code> directory.
Here is an example of script that we will explain, and where you need to change some paths...</p>
<pre><code>#!/bin/bash
## SLURM CONFIG ##

#SBATCH --job-name=amrfinder
#SBATCH --output=%x.%j.out
#SBATCH --cpus-per-task 4
#SBATCH --time=24:00:00
#SBATCH -p SELECTED_PARTITION
#SBATCH --mail-type=FAIL,END
#SBATCH --mem-per-cpu=4G

###
HELP=&quot;USAGE: amrfinderplus.sh contigs.fasta&quot;
# If we didn't get any arguments, print help and exit
if [[ $# &lt; 1 ]]
then
    echo &quot;$HELP&quot;
    exit 0
fi

#variables
suffix=&quot;.fasta&quot;
prefix1=&quot;${1%$suffix}&quot;

##
module load system/singularity/3.6.0

singularity run path/to/amrfinder_container \
  amrfinder --nucleotide ${1} -o ${prefix1}_AMRfinder.txt --organism Klebsiella --plus --mutation_all ${prefix1}_AMRfinder_all_mut.txt
</code></pre>
<p>Once your script seems ready, exit the container, and the <code>srun</code> if you are still on the node.
Then you are ready to loop on the unicycler fasta files for running the script with <code>sbatch</code></p>
<p>First, we can test it on one strain:</p>
<pre><code class="language-bash">sbatch ../scripts/amrfinder_script.sh K1_unicycler_scaffolds.fasta
</code></pre>
<p>If the test has worked, you can run the loop:</p>
<pre><code class="language-bash">for i in K*_unicycler_scaffolds.fasta; \
  do sbatch ../scripts/amrfinder_script.sh $i;\
  done
</code></pre>
<h2 id="card-rgi">CARD RGI<a class="headerlink" href="#card-rgi" title="Permanent link">#</a></h2>
<p>Now that we feel more familiar with preparing SLURM scripts, we will make <code>card_script.sh</code> for running CARD RGI on all samples.</p>
<pre><code>#!/bin/bash
## SLURM CONFIG ##

#SBATCH --job-name=card
#SBATCH --output=%x.%j.out
#SBATCH --cpus-per-task 4
#SBATCH --time=24:00:00
#SBATCH -p highmemplus
#SBATCH --mail-type=FAIL,END
#SBATCH --mem-per-cpu=4G

###

HELP=&quot;
USAGE: card_script.sh contigs.fasta
&quot;
# If we didn't get any arguments, print help and exit
if [[ $# &lt; 1 ]]
then
    echo &quot;$HELP&quot;
    exit 0
fi

#variables
suffix=&quot;.fasta&quot;
prefix1=&quot;${1%$suffix}&quot;

##
module load system/singularity/3.6.0

# loading the CARD database
singularity run /path/to/containers/rgi_6.0.0--pyha8f3691_0.sif \
  rgi load --card_json /path/to/databases/CARD/card.json

# Running RGI Main =&gt; output json for Heatmap plotting
singularity run /path/to/containers/rgi_6.0.0--pyha8f3691_0.sif \
  rgi main -i ${1} -o ${prefix1}_RGI_main --debug -a BLAST -d wgs -n 4

# Preparing tabular file for import to excel or R
singularity run /path/to/containers/rgi_6.0.0--pyha8f3691_0.sif \
  rgi tab -i ${prefix1}_RGI_main.json

rm ${1}.temp.*
rm ${1}.temp
</code></pre>
<p>Make a test run on one sample before running the loop.</p>
<p>If the test has worked, you can run the loop:</p>
<pre><code class="language-bash">for i in K*_unicycler_scaffolds.fasta; \
  do sbatch ../scripts/card_script.sh $i;\
  done
</code></pre>
<p>Once we have all the json files from CARD RGI, we will place them in the same directory.</p>
<pre><code class="language-bash">mkdir card_json
mv *_RGI_main.json card_json
</code></pre>
<p>After that, we can run the RGI program for making the heatmap clustering
Run the <code>srun</code>command if you are not already on the computing node:</p>
<pre><code class="language-bash">srun -p SELECTED_PARTITION --cpus-per-task 2 --pty bash -i
</code></pre>
<p>Run RGI heatmap:</p>
<pre><code class="language-bash">module load system/singularity/3.6.0

singularity run path/to/card_container

#Generate a heat map from pre-compiled RGI main JSON files, samples clustered by similarity of resistome and AMR genes organized by Drug Class
rgi heatmap -i card_json -cat drug_class -o hm_drug_class -clus samples
</code></pre>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>You can find other ways to cluster/classify the resistance genes, as the following examples:</p>
</div>
<pre><code class="language-bash">#Generate a heat map from pre-compiled RGI main JSON files, samples clustered by similarity of resistome and AMR genes organized by resistance resistance_mechanism
rgi heatmap -i card_json -cat resistance_mechanism -o hm_resistance_mechanism -clus samples
#Generate a heat map from pre-compiled RGI main JSON files, samples clustered by similarity of resistome and AMR genes organized by AMR gene family
rgi heatmap -i card_json -cat gene_family -o hm_genefamily_samples -clus samples

#Generate a heat map from pre-compiled RGI main JSON files, samples clustered by similarity of resistome (with histogram used for abundance of identical resistomes) and AMR genes organized by distribution among samples:
rgi heatmap -i card_json -o cluster_both_frequency -f -clus both
</code></pre>
<p>Then you can <code>scp</code> all the pictures on you computer and inspect.</p>

			<br>
			






<div class="row pt-2 pb-4">
	<div class="col-sm text-center text-sm-left">
		<a href="../mlst/" class="btn btn-sm btn-outline-dark">
			<i class="fa fa-chevron-left" aria-hidden="true"></i>
			Previous
		</a>
		<a href="../mlst/" class="btn btn-sm btn-link">
			MLST
		</a>
	</div>
	

	
	<div class="col-sm text-center text-sm-right">
		<a href="../plasmids/" class="btn btn-sm btn-link">
			Plasmids
		</a>
		<a href="../plasmids/" class="btn btn-sm btn-outline-dark">
			Next
			<i class="fa fa-chevron-right" aria-hidden="true"></i>
		</a>
	</div>
	
</div>

			<br>
		</div>

		<footer class="container-fluid wm-page-content text-center small">
			<p>
			<a href="https://github.com/jhayer/bacterial_genomics_training/edit/master/docs/args.md" target="_blank">Edit on jhayer/bacterial_genomics_training</a>
			</p>
			<details>
				<summary>
					Documentation built with <a href="http://www.mkdocs.org/" target="_blank">MkDocs</a> using  <a href="https://github.com/Siphalor/mkdocs-custommill" target="_blank">CustomMill</a>.
				</summary>
				<h6 class="mt-2">Additional Licenses</h6>
				<ul><li>
						CustomMill is based on the <a href="https://github.com/gristlabs/mkdocs-windmill" target="_blank">Windmill</a> theme by Grist Labs, licensed under the <a href="https://github.com/gristlabs/mkdocs-windmill/blob/master/LICENSE" target="_blank">MIT License</a>
					</li>
					<li>
						CustomMill includes <a href="https://getbootstrap.com" target="_blank">Bootstrap</a> version 5.1.3. This is available under the <a href="https://github.com/twbs/bootstrap/blob/master/LICENSE" target="_blank">MIT License</a>.
					</li>
					<li>
						Syntax highlighting is implemented via <a href="https://highlightjs.org" target="_blank">highlight.js</a> licensed under the BSD 3-Clause License and the included github styling made by Vasily Polovnyov.
					</li>
					<li>
						Icons on this site are from <a href="https://fontawesome.com/" target="_blank">Font Awesome</a> which is licensed under the <a href="https://fontawesome.com/license/free" target="_blank">Font Awesome Free License</a>.
					</li></ul>
				<p>Built at 2022-11-23 15:03:06 UTC</p>
			</details>
		</footer>

		
	</body>
</html>