<!DOCTYPE html>
<html lang="en">
	<head>
		
		
		<meta charset="utf-8">
		<meta http-equiv="X-UA-Compatible" content="IE=edge">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		
		<meta name="author" content="Juliette Hayer and Lokman Galal">
		<link rel="canonical" href="https://jhayer.github.io/bacterial_genomics_training/plasmids/">
		<link rel="shortcut icon" href="../img/favicon.ico">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
		<title>Plasmids - Bioinformatics Tutorials for Bacterial Genomics</title>
		<link href="../bootstrap/dist/css/bootstrap.min.css" rel="stylesheet">
		<link href="../css/font-awesome-5.12.2-all.min.css" rel="stylesheet">
		<link href="../css/base.css" rel="stylesheet">
		<link rel="stylesheet" href="../css/highlight-github.css">
		<script src="../bootstrap/dist/js/bootstrap.min.js"></script>
		<script src="../js/highlight.pack.js"></script>
		
		<script>
			var base_url = '..';
			var home_url = '../usage';
			var is_outer_page = false;
			
			var pageToc = [
				{title: "Plasmids detection", url: "#_top", children: [
						
				{title: "Prepare our computing environment", url: "#_top", children: [
				]},
				{title: "Practical", url: "#practical", children: [
				]},
				{title: "PlasmidFinder", url: "#plasmidfinder", children: [
				]},
				{title: "Platon", url: "#platon", children: [
				]},
				]},
			];

		</script>
		<script src="../js/base.js"></script> 
	</head>

	<body class="inner-page">
		

		<div class="container-fluid wm-page-content">
			<a name="_top" aria-hidden="true"></a>
			






<div class="row pt-2 pb-4">
	<div class="col-sm text-center text-sm-left">
		<a href="../args/" class="btn btn-sm btn-outline-dark">
			<i class="fa fa-chevron-left" aria-hidden="true"></i>
			Previous
		</a>
		<a href="../args/" class="btn btn-sm btn-link">
			Antimicrobial Resistance Genes
		</a>
	</div>
	

	
	<div class="col-sm text-center text-sm-right">
		<a href="../annotation/" class="btn btn-sm btn-link">
			Annotation
		</a>
		<a href="../annotation/" class="btn btn-sm btn-outline-dark">
			Next
			<i class="fa fa-chevron-right" aria-hidden="true"></i>
		</a>
	</div>
	
</div>

			

			<h1 id="plasmids-detection">Plasmids detection<a class="headerlink" href="#plasmids-detection" title="Permanent link">#</a></h1>
<p>We will now search for plasmids features in the assembled scaffolds/contigs to detect which ones could have originated from a plasmid.</p>
<p>For this, we will use 2 different tools:</p>
<ul>
<li>
<p><a href="https://cge.food.dtu.dk/services/PlasmidFinder/">PlasmidFinder</a>, developed at the <a href="http://www.genomicepidemiology.org/services/">Center for Genomic Epidemiology</a>, like many other tools for bacterial genomics analysis. It searched for known plasmids and provide the replicon type.</p>
</li>
<li>
<p><a href="https://github.com/oschwengers/platon">Platon</a>, a tool that detects plasmid-borne contigs within bacterial draft (meta) genomes assemblies. Therefore, Platon analyzes the distribution bias of protein-coding gene families among chromosomes and plasmids. This analysis is complemented by comprehensive contig characterizations followed by heuristic filters.
Platon depends on a custom database based on MPS, RDS, RefSeq Plasmid database, PlasmidFinder db as well as manually curated MOB HMM models from MOBscan, custom conjugation and replication HMM models and oriT sequences from MOB-suite.</p>
</li>
</ul>
<h2 id="prepare-our-computing-environment">Prepare our computing environment<a class="headerlink" href="#prepare-our-computing-environment" title="Permanent link">#</a></h2>
<p>We will first run the appropriate <code>srun</code> command to book the computing cores (cpus) on the cluster.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>You need to ask the teacher which partition to use !</p>
</div>
<pre><code class="language-bash">srun -p SELECTED_PARTITION --cpus-per-task 2 --pty bash -i
</code></pre>
<p>You are now on a computing node, with computing 2 cpus reserved for you. That way, you can run commands interactively.</p>
<p>If you want to exit the <code>srun</code> interactive mode, press CTRL+D or type <code>exit</code></p>
<h2 id="practical">Practical<a class="headerlink" href="#practical" title="Permanent link">#</a></h2>
<p>First, you need to identify in your results directory the fasta file corresponding to the unicycler hybrid assembly.</p>
<pre><code class="language-bash">cd results
ls -l
</code></pre>
<h2 id="plasmidfinder">PlasmidFinder<a class="headerlink" href="#plasmidfinder" title="Permanent link">#</a></h2>
<p>We will use the PlasmidFinder container that has all tools needed for running PlasmidFinder</p>
<pre><code class="language-bash">module load system/singularity/3.6.0

singularity shell path/to/plasmidfiner_container

mkdir K2_plasmidfinder

plasmidfinder.py -i K2_unicycler_scaffolds.fasta -o K2_plasmidfinder -p /path/to/PF_DB -mp blastn -x
</code></pre>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Ask the teacher for the PlasmidFinder database path and for the PlasmidFinder container path.</p>
</div>
<p>Once finished, you can exit the container by typing <code>exit</code> or press CTRL+D</p>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>Inspect the output files within the plasmidfinder output directory and comment.
How many potential contigs of plasmidic origin were identified ?</p>
</div>
<h2 id="platon">Platon<a class="headerlink" href="#platon" title="Permanent link">#</a></h2>
<p>We will use a second tool for detecting sequences of plasmidic origin: <a href="https://github.com/oschwengers/platon">Platon</a></p>
<pre><code class="language-bash">module load system/singularity/3.6.0

singularity shell path/to/platon_container

platon --db path/to/databases/platon/db/ -t 4 -o K2_unicycler_platon_accu K2_unicycler_scaffolds.fasta
</code></pre>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Ask the teacher for the Platon database path and for the Platon container path.</p>
</div>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>Inspect the output files within the Platon output directory and comment.
Compare with results obtained with PlasmidFinder</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can put the Platon json file in <a href="https://jsoneditoronline.org">Json Editor online</a> for visualising the hits</p>
</div>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>Try to run the tool on all other Unicycler assemblies by using a loop</p>
</div>

			<br>
			






<div class="row pt-2 pb-4">
	<div class="col-sm text-center text-sm-left">
		<a href="../args/" class="btn btn-sm btn-outline-dark">
			<i class="fa fa-chevron-left" aria-hidden="true"></i>
			Previous
		</a>
		<a href="../args/" class="btn btn-sm btn-link">
			Antimicrobial Resistance Genes
		</a>
	</div>
	

	
	<div class="col-sm text-center text-sm-right">
		<a href="../annotation/" class="btn btn-sm btn-link">
			Annotation
		</a>
		<a href="../annotation/" class="btn btn-sm btn-outline-dark">
			Next
			<i class="fa fa-chevron-right" aria-hidden="true"></i>
		</a>
	</div>
	
</div>

			<br>
		</div>

		<footer class="container-fluid wm-page-content text-center small">
			<p>
			<a href="https://github.com/jhayer/bacterial_genomics_training/edit/master/docs/plasmids.md" target="_blank">Edit on jhayer/bacterial_genomics_training</a>
			</p>
			<details>
				<summary>
					Documentation built with <a href="http://www.mkdocs.org/" target="_blank">MkDocs</a> using  <a href="https://github.com/Siphalor/mkdocs-custommill" target="_blank">CustomMill</a>.
				</summary>
				<h6 class="mt-2">Additional Licenses</h6>
				<ul><li>
						CustomMill is based on the <a href="https://github.com/gristlabs/mkdocs-windmill" target="_blank">Windmill</a> theme by Grist Labs, licensed under the <a href="https://github.com/gristlabs/mkdocs-windmill/blob/master/LICENSE" target="_blank">MIT License</a>
					</li>
					<li>
						CustomMill includes <a href="https://getbootstrap.com" target="_blank">Bootstrap</a> version 5.1.3. This is available under the <a href="https://github.com/twbs/bootstrap/blob/master/LICENSE" target="_blank">MIT License</a>.
					</li>
					<li>
						Syntax highlighting is implemented via <a href="https://highlightjs.org" target="_blank">highlight.js</a> licensed under the BSD 3-Clause License and the included github styling made by Vasily Polovnyov.
					</li>
					<li>
						Icons on this site are from <a href="https://fontawesome.com/" target="_blank">Font Awesome</a> which is licensed under the <a href="https://fontawesome.com/license/free" target="_blank">Font Awesome Free License</a>.
					</li></ul>
				<p>Built at 2022-11-18 15:52:35 UTC</p>
			</details>
		</footer>

		
	</body>
</html>