<!DOCTYPE html>
<html lang="en">
	<head>
		
		
		<meta charset="utf-8">
		<meta http-equiv="X-UA-Compatible" content="IE=edge">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		
		<meta name="author" content="Juliette Hayer and Lokman Galal">
		<link rel="canonical" href="https://jhayer.github.io/bacterial_genomics_training/basecalling/">
		<link rel="shortcut icon" href="../img/favicon.ico">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
		<title>Long reads Basecalling - Bioinformatics Tutorials for Bacterial Genomics</title>
		<link href="../bootstrap/dist/css/bootstrap.min.css" rel="stylesheet">
		<link href="../css/font-awesome-5.12.2-all.min.css" rel="stylesheet">
		<link href="../css/base.css" rel="stylesheet">
		<link rel="stylesheet" href="../css/highlight-github.css">
		<script src="../bootstrap/dist/js/bootstrap.min.js"></script>
		<script src="../js/highlight.pack.js"></script>
		
		<script>
			var base_url = '..';
			var home_url = '../usage';
			var is_outer_page = false;
			
			var pageToc = [
				{title: "BASE CALLING", url: "#_top", children: [
						
				{title: "objective", url: "#_top", children: [
				]},
				{title: "environment", url: "#environment", children: [
				]},
				{title: "software", url: "#software", children: [
				]},
				{title: "location of input files and how to enter them in the command line", url: "#location-of-input-files-and-how-to-enter-them-in-the-command-line", children: [
				]},
				{title: "location of output files and how to set it in the command line", url: "#location-of-output-files-and-how-to-set-it-in-the-command-line", children: [
				]},
				{title: "other required files and how to set them in the command line", url: "#other-required-files-and-how-to-set-them-in-the-command-line", children: [
				]},
				{title: "command line to launch guppy", url: "#command-line-to-launch-guppy", children: [
				]},
				{title: "concatenate fastq files passing quality evaluation", url: "#concatenate-fastq-files-passing-quality-evaluation", children: [
				]},
				]},
			];

		</script>
		<script src="../js/base.js"></script> 
	</head>

	<body class="inner-page">
		

		<div class="container-fluid wm-page-content">
			<a name="_top" aria-hidden="true"></a>
			






<div class="row pt-2 pb-4">
	<div class="col-sm text-center text-sm-left">
		<a href="../file_formats/" class="btn btn-sm btn-outline-dark">
			<i class="fa fa-chevron-left" aria-hidden="true"></i>
			Previous
		</a>
		<a href="../file_formats/" class="btn btn-sm btn-link">
			File formats
		</a>
	</div>
	

	
	<div class="col-sm text-center text-sm-right">
		<a href="../qc/" class="btn btn-sm btn-link">
			Short Reads Quality Control and Trimming
		</a>
		<a href="../qc/" class="btn btn-sm btn-outline-dark">
			Next
			<i class="fa fa-chevron-right" aria-hidden="true"></i>
		</a>
	</div>
	
</div>

			

			<h1 id="base-calling">BASE CALLING<a class="headerlink" href="#base-calling" title="Permanent link">#</a></h1>
<p>&nbsp;</p>
<h2 id="objective">objective<a class="headerlink" href="#objective" title="Permanent link">#</a></h2>
<p>Base calling is the process of translating the electronic raw signal (fast5 format) of the sequencer into bases (fastq format).
&nbsp;</p>
<p>&nbsp;</p>
<h2 id="environment">environment<a class="headerlink" href="#environment" title="Permanent link">#</a></h2>
<p>This process will be carried out on the i-Trop computing cluster.
&nbsp;</p>
<p>&nbsp;</p>
<h2 id="software">software<a class="headerlink" href="#software" title="Permanent link">#</a></h2>
<p>guppy
&nbsp;</p>
<p><span style="color:green"><strong>QUESTION 1: What is the lasted version of guppy installed on the i-Trop computing cluster?</strong></span>
&nbsp;</p>
<p>&nbsp;</p>
<h2 id="location-of-input-files-and-how-to-enter-them-in-the-command-line">location of input files and how to enter them in the command line<a class="headerlink" href="#location-of-input-files-and-how-to-enter-them-in-the-command-line" title="Permanent link">#</a></h2>
<p>long_reads_analyses/fast5/</p>
<p>-i [ --input_path ]
    Path to input files.
&nbsp;</p>
<p>&nbsp;</p>
<h2 id="location-of-output-files-and-how-to-set-it-in-the-command-line">location of output files and how to set it in the command line<a class="headerlink" href="#location-of-output-files-and-how-to-set-it-in-the-command-line" title="Permanent link">#</a></h2>
<p>long_reads_analyses/guppy_output/</p>
<p>-s [ --save_path ]
    Path to save output files.
&nbsp;</p>
<p>&nbsp;</p>
<h2 id="other-required-files-and-how-to-set-them-in-the-command-line">other required files and how to set them in the command line<a class="headerlink" href="#other-required-files-and-how-to-set-them-in-the-command-line" title="Permanent link">#</a></h2>
<p>Entering information on which flowcell and library preparation kit was used for sequencing is mandatory for launching guppy.
This data is found in configuration files. Each library preparation kit/flowcell pair corresponds to a unique configuration file. </p>
<p>-c [ --config ]
    Configuration file for application.</p>
<p>To list available configuration files enter <code>guppy_basecaller ––print_workflows</code> (to be entered manually!).
The provided data was prepared using the SQK-LSK108 library preparation kit that was sequenced on a MIN106 flowcell.
&nbsp;</p>
<p><span style="color:green"><strong>QUESTION 2: What is the name of the configuration file guppy needs to basecall the tutorial data?</strong></span>
&nbsp;</p>
<p>&nbsp;</p>
<h2 id="command-line-to-launch-guppy">command line to launch guppy<a class="headerlink" href="#command-line-to-launch-guppy" title="Permanent link">#</a></h2>
<pre><code>guppy_basecaller \
-i */fast5 \
-s */guppy_out \
-c dna_*.cfg \
--num_callers 1 --cpu_threads_per_caller 1
</code></pre>
<p>Let the run continue and go to the directory containing precompiled output files.
It contains pass and fail directories, each containing fastq files.</p>
<p>Check the sequencing_summary file.
&nbsp;</p>
<p><span style="color:green"><strong>QUESTION 3: What is the minimal threshold for the mean quality score (mean qscore) to pass quality evaluation?</strong></span>
&nbsp;</p>
<p>&nbsp;</p>
<h2 id="concatenate-fastq-files-passing-quality-evaluation">concatenate fastq files passing quality evaluation<a class="headerlink" href="#concatenate-fastq-files-passing-quality-evaluation" title="Permanent link">#</a></h2>
<p>guppy output is often composed of several fastq files for a single sample that must be concatenated. 
<code>cat *.fastq &gt; sample1.fastq</code></p>

			<br>
			






<div class="row pt-2 pb-4">
	<div class="col-sm text-center text-sm-left">
		<a href="../file_formats/" class="btn btn-sm btn-outline-dark">
			<i class="fa fa-chevron-left" aria-hidden="true"></i>
			Previous
		</a>
		<a href="../file_formats/" class="btn btn-sm btn-link">
			File formats
		</a>
	</div>
	

	
	<div class="col-sm text-center text-sm-right">
		<a href="../qc/" class="btn btn-sm btn-link">
			Short Reads Quality Control and Trimming
		</a>
		<a href="../qc/" class="btn btn-sm btn-outline-dark">
			Next
			<i class="fa fa-chevron-right" aria-hidden="true"></i>
		</a>
	</div>
	
</div>

			<br>
		</div>

		<footer class="container-fluid wm-page-content text-center small">
			<p>
			<a href="https://github.com/jhayer/bacterial_genomics_training/edit/master/docs/basecalling.md" target="_blank">Edit on jhayer/bacterial_genomics_training</a>
			</p>
			<details>
				<summary>
					Documentation built with <a href="http://www.mkdocs.org/" target="_blank">MkDocs</a> using  <a href="https://github.com/Siphalor/mkdocs-custommill" target="_blank">CustomMill</a>.
				</summary>
				<h6 class="mt-2">Additional Licenses</h6>
				<ul><li>
						CustomMill is based on the <a href="https://github.com/gristlabs/mkdocs-windmill" target="_blank">Windmill</a> theme by Grist Labs, licensed under the <a href="https://github.com/gristlabs/mkdocs-windmill/blob/master/LICENSE" target="_blank">MIT License</a>
					</li>
					<li>
						CustomMill includes <a href="https://getbootstrap.com" target="_blank">Bootstrap</a> version 5.1.3. This is available under the <a href="https://github.com/twbs/bootstrap/blob/master/LICENSE" target="_blank">MIT License</a>.
					</li>
					<li>
						Syntax highlighting is implemented via <a href="https://highlightjs.org" target="_blank">highlight.js</a> licensed under the BSD 3-Clause License and the included github styling made by Vasily Polovnyov.
					</li>
					<li>
						Icons on this site are from <a href="https://fontawesome.com/" target="_blank">Font Awesome</a> which is licensed under the <a href="https://fontawesome.com/license/free" target="_blank">Font Awesome Free License</a>.
					</li></ul>
				<p>Built at 2022-10-17 13:22:35 UTC</p>
			</details>
		</footer>

		
	</body>
</html>