<!DOCTYPE html>
<html lang="en">
	<head>
		
		
		<meta charset="utf-8">
		<meta http-equiv="X-UA-Compatible" content="IE=edge">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		
		<meta name="author" content="Juliette Hayer and Lokman Galal">
		<link rel="canonical" href="https://jhayer.github.io/bacterial_genomics_training/qc/">
		<link rel="shortcut icon" href="../img/favicon.ico">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
		<title>Short Reads Quality Control and Trimming - Bioinformatics Tutorials for Bacterial Genomics</title>
		<link href="../bootstrap/dist/css/bootstrap.min.css" rel="stylesheet">
		<link href="../css/font-awesome-5.12.2-all.min.css" rel="stylesheet">
		<link href="../css/base.css" rel="stylesheet">
		<link rel="stylesheet" href="../css/highlight-github.css">
		<script src="../bootstrap/dist/js/bootstrap.min.js"></script>
		<script src="../js/highlight.pack.js"></script>
		
		<script>
			var base_url = '..';
			var home_url = '../usage';
			var is_outer_page = false;
			
			var pageToc = [
				{title: "Quality Control and Trimming", url: "#_top", children: [
						
				{title: "Lecture", url: "#_top", children: [
				]},
				{title: "Practical", url: "#practical", children: [
				]},
				{title: "Downloading the data", url: "#downloading-the-data", children: [
				]},
				{title: "Working Directory", url: "#working-directory", children: [
				]},
				{title: "FastQC", url: "#fastqc", children: [
				]},
				{title: "Scythe", url: "#scythe", children: [
				]},
				{title: "Sickle", url: "#sickle", children: [
				]},
				{title: "FastQC again", url: "#fastqc-again", children: [
				]},
				{title: "MultiQC", url: "#multiqc", children: [
				]},
				]},
			];

		</script>
		<script src="../js/base.js"></script> 
	</head>

	<body class="inner-page">
		

		<div class="container-fluid wm-page-content">
			<a name="_top" aria-hidden="true"></a>
			






<div class="row pt-2 pb-4">
	<div class="col-sm text-center text-sm-left">
		<a href="../basecalling/" class="btn btn-sm btn-outline-dark">
			<i class="fa fa-chevron-left" aria-hidden="true"></i>
			Previous
		</a>
		<a href="../basecalling/" class="btn btn-sm btn-link">
			Long reads Basecalling
		</a>
	</div>
	

	
	<div class="col-sm text-center text-sm-right">
		<a href="../mkdocs_usage/" class="btn btn-sm btn-link">
			Mkdocs usage
		</a>
		<a href="../mkdocs_usage/" class="btn btn-sm btn-outline-dark">
			Next
			<i class="fa fa-chevron-right" aria-hidden="true"></i>
		</a>
	</div>
	
</div>

			

			<h1 id="quality-control-and-trimming">Quality Control and Trimming<a class="headerlink" href="#quality-control-and-trimming" title="Permanent link">#</a></h1>
<h2 id="lecture">Lecture<a class="headerlink" href="#lecture" title="Permanent link">#</a></h2>
<p><br></p>
<iframe src="https://docs.google.com/presentation/d/e/2PACX-1vR7ZN5Gc-3q5DR8CWQgyHhjUIc-6uKJnB48lDVMB7tWrt4gpZFipbJRVWfaqdxSw9n_WnMfXoFMZXds/embed?start=false&loop=false&delayms=3000" frameborder="0" width="480" height="389" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe>

<h2 id="practical">Practical<a class="headerlink" href="#practical" title="Permanent link">#</a></h2>
<p>In this practical you will learn to import, view and check the quality of raw high thoughput sequencing sequencing data.</p>
<p>The first dataset you will be working with is from an Illumina MiSeq dataset.
The sequenced organism is an enterohaemorrhagic E. coli (EHEC) of the serotype O157, a potentially fatal gastrointestinal pathogen.
The sequenced bacterium was part of an outbreak investigation in the St. Louis area, USA in 2011.
The sequencing was done as paired-end 2x150bp.</p>
<h2 id="downloading-the-data">Downloading the data<a class="headerlink" href="#downloading-the-data" title="Permanent link">#</a></h2>
<p>The raw data were deposited at the European Nucleotide Archive, under the accession number SRR957824.
You could go to the ENA <a href="http://www.ebi.ac.uk/ena">website</a> and search for the run with the accession SRR957824.</p>
<p>However these files contain about 3 million reads and are therefore quite big.
We are only going to use a subset of the original dataset for this tutorial.</p>
<p>First create a <code>data/</code> directory in your home folder</p>
<pre><code class="language-bash">mkdir ~/data
</code></pre>
<p>now let's download the subset</p>
<pre><code class="language-bash">cd ~/data
curl -O -J -L https://osf.io/shqpv/download
curl -O -J -L https://osf.io/9m3ch/download
</code></pre>
<p>Let’s make sure we downloaded all of our data using md5sum.</p>
<pre><code class="language-bash">md5sum SRR957824_500K_R1.fastq.gz SRR957824_500K_R2.fastq.gz
</code></pre>
<p>you should see this</p>
<pre><code>1e8cf249e3217a5a0bcc0d8a654585fb  SRR957824_500K_R1.fastq.gz
70c726a31f05f856fe942d727613adb7  SRR957824_500K_R2.fastq.gz
</code></pre>
<p>and now look at the file names and their size</p>
<pre><code class="language-bash">ls -l
</code></pre>
<pre><code>total 97M
-rw-r--r-- 1 hadrien 48M Nov 19 18:44 SRR957824_500K_R1.fastq.gz
-rw-r--r-- 1 hadrien 50M Nov 19 18:53 SRR957824_500K_R2.fastq.gz
</code></pre>
<p>There are 500 000 paired-end reads taken randomly from the original data</p>
<p>One last thing before we get to the quality control: those files are writeable.
By default, UNIX makes things writeable by the file owner.
This poses an issue with creating typos or errors in raw data.
We fix that before going further</p>
<pre><code class="language-bash">chmod u-w *
</code></pre>
<h2 id="working-directory">Working Directory<a class="headerlink" href="#working-directory" title="Permanent link">#</a></h2>
<p>First we make a work directory: a directory where we can play around with a copy of the data without messing with the original</p>
<pre><code class="language-bash">mkdir ~/work
cd ~/work
</code></pre>
<p>Now we make a link of the data in our working directory</p>
<pre><code class="language-bash">ln -s ~/data/* .
</code></pre>
<p>The files that we've downloaded are FASTQ files. Take a look at one of them with</p>
<pre><code class="language-bash">zless SRR957824_500K_R1.fastq.gz
</code></pre>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
</div>
<p>Use the spacebar to scroll down, and type ‘q’ to exit ‘less’</p>
<p>You can read more on the FASTQ format in the <a href="../file_formats/">File Formats</a> lesson.</p>
<div class="admonition question">
<p class="admonition-title">Question</p>
</div>
<p>Where does the filename come from?</p>
<div class="admonition question">
<p class="admonition-title">Question</p>
</div>
<p>Why are there 1 and 2 in the file names?</p>
<h2 id="fastqc">FastQC<a class="headerlink" href="#fastqc" title="Permanent link">#</a></h2>
<p>To check the quality of the sequence data we will use a tool called FastQC.</p>
<p>FastQC has a graphical interface and can be downloaded and run on a Windows or Linux computer without installation.
It is available <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">here</a>.</p>
<p>However, FastQC is also available as a command line utility on the training server you are using.
To run FastQC on our two files</p>
<pre><code class="language-bash">fastqc SRR957824_500K_R1.fastq.gz SRR957824_500K_R2.fastq.gz
</code></pre>
<p>and look what FastQC has produced</p>
<pre><code>ls *fastqc*
</code></pre>
<p>For each file, FastQC has produced both a .zip archive containing all the plots, and a html report.</p>
<p>Download and open the html files with your favourite web browser.</p>
<p>Alternatively you can look a these copies of them:</p>
<ul>
<li><a href="data/fastqc/SRR957824_500K_R1_fastqc.html">SRR957824_500K_R1_fastqc.html</a></li>
<li><a href="data/fastqc/SRR957824_500K_R2_fastqc.html">SRR957824_500K_R2_fastqc.html</a></li>
</ul>
<div class="admonition question">
<p class="admonition-title">Question</p>
</div>
<p>What should you pay attention to in the FastQC report?</p>
<div class="admonition question">
<p class="admonition-title">Question</p>
</div>
<p>Which file is of better quality?</p>
<p>Pay special attention to the per base sequence quality and sequence length distribution.
Explanations for the various quality modules can be found <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/">here</a>.
Also, have a look at examples of a <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html">good</a> and a <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html">bad</a> illumina read set for comparison.</p>
<p>You will note that the reads in your uploaded dataset have fairly poor quality (&lt;20) towards the end. There are also outlier reads that have very poor quality for most of the second half of the reads.</p>
<h2 id="scythe">Scythe<a class="headerlink" href="#scythe" title="Permanent link">#</a></h2>
<p>Now we'll do some trimming!</p>
<p>Scythe uses a Naive Bayesian approach to classify contaminant substrings in sequence reads.
It considers quality information, which can make it robust in picking out 3'-end adapters, which often include poor quality bases.</p>
<p>The first thing we need is the adapters to trim off</p>
<pre><code class="language-bash">curl -O -J -L https://osf.io/v24pt/download
</code></pre>
<p>Now we run scythe on both our read files</p>
<pre><code class="language-bash">scythe -a adapters.fasta SRR957824_500K_R1.fastq.gz -o SRR957824_adapt_R1.fastq 
scythe -a adapters.fasta SRR957824_500K_R2.fastq.gz -o SRR957824_adapt_R2.fastq 
</code></pre>
<div class="admonition question">
<p class="admonition-title">Question</p>
</div>
<p>What adapters do you use?</p>
<h2 id="sickle">Sickle<a class="headerlink" href="#sickle" title="Permanent link">#</a></h2>
<p>Most modern sequencing technologies produce reads that have deteriorating quality towards the 3'-end and some towards the 5'-end as well.
Incorrectly called bases in both regions negatively impact assembles, mapping, and downstream bioinformatics analyses.</p>
<p>We will trim each read individually down to the good quality part to keep the bad part from interfering with downstream applications.</p>
<p>To do so, we will use sickle. Sickle is a tool that uses sliding windows along with quality and length thresholds to determine when quality is sufficiently low to trim the 3'-end of reads and also determines when the quality is sufficiently high enough to trim the 5'-end of reads. It will also discard reads based upon a length threshold.</p>
<p>To run sickle</p>
<pre><code class="language-bash">sickle pe -f SRR957824_adapt_R1.fastq -r SRR957824_adapt_R2.fastq \
    -t sanger -o SRR957824_trimmed_R1.fastq -p SRR957824_trimmed_R2.fastq \
    -s /dev/null -q 25
</code></pre>
<p>which should output something like</p>
<pre><code>PE forward file: SRR957824_trimmed_R1.fastq
PE reverse file: SRR957824_trimmed_R2.fastq

Total input FastQ records: 1000000 (500000 pairs)

FastQ paired records kept: 834570 (417285 pairs)
FastQ single records kept: 13263 (from PE1: 11094, from PE2: 2169)
FastQ paired records discarded: 138904 (69452 pairs)
FastQ single records discarded: 13263 (from PE1: 2169, from PE2: 11094)
</code></pre>
<h2 id="fastqc-again">FastQC again<a class="headerlink" href="#fastqc-again" title="Permanent link">#</a></h2>
<p>Run fastqc again on the filtered reads</p>
<pre><code class="language-bash">fastqc SRR957824_trimmed_R1.fastq SRR957824_trimmed_R2.fastq
</code></pre>
<p>and look at the reports</p>
<ul>
<li><a href="data/fastqc/SRR957824_trimmed_R1_fastqc.html">SRR957824_trimmed_R1_fastqc.html</a></li>
<li><a href="data/fastqc/SRR957824_trimmed_R2_fastqc.html">SRR957824_trimmed_R2_fastqc.html</a></li>
</ul>
<h2 id="multiqc">MultiQC<a class="headerlink" href="#multiqc" title="Permanent link">#</a></h2>
<p><a href="http://multiqc.info">MultiQC</a> is a tool that aggreagtes results from several popular QC bioinformatics software into one html report.</p>
<p>Let's run MultiQC in our current directory</p>
<pre><code class="language-bash">multiqc .
</code></pre>
<p>You can download the report or view it by clicking on the link below</p>
<ul>
<li><a href="data/fastqc/multiqc_report.html">multiqc_report.html</a></li>
</ul>
<div class="admonition question">
<p class="admonition-title">Question</p>
</div>
<p>What did the trimming do to the per-base sequence quality, the per sequence quality scores and the sequence length distribution?</p>

			<br>
			






<div class="row pt-2 pb-4">
	<div class="col-sm text-center text-sm-left">
		<a href="../basecalling/" class="btn btn-sm btn-outline-dark">
			<i class="fa fa-chevron-left" aria-hidden="true"></i>
			Previous
		</a>
		<a href="../basecalling/" class="btn btn-sm btn-link">
			Long reads Basecalling
		</a>
	</div>
	

	
	<div class="col-sm text-center text-sm-right">
		<a href="../mkdocs_usage/" class="btn btn-sm btn-link">
			Mkdocs usage
		</a>
		<a href="../mkdocs_usage/" class="btn btn-sm btn-outline-dark">
			Next
			<i class="fa fa-chevron-right" aria-hidden="true"></i>
		</a>
	</div>
	
</div>

			<br>
		</div>

		<footer class="container-fluid wm-page-content text-center small">
			<p>
			<a href="https://github.com/jhayer/bacterial_genomics_training/edit/master/docs/qc.md" target="_blank">Edit on jhayer/bacterial_genomics_training</a>
			</p>
			<details>
				<summary>
					Documentation built with <a href="http://www.mkdocs.org/" target="_blank">MkDocs</a> using  <a href="https://github.com/Siphalor/mkdocs-custommill" target="_blank">CustomMill</a>.
				</summary>
				<h6 class="mt-2">Additional Licenses</h6>
				<ul><li>
						CustomMill is based on the <a href="https://github.com/gristlabs/mkdocs-windmill" target="_blank">Windmill</a> theme by Grist Labs, licensed under the <a href="https://github.com/gristlabs/mkdocs-windmill/blob/master/LICENSE" target="_blank">MIT License</a>
					</li>
					<li>
						CustomMill includes <a href="https://getbootstrap.com" target="_blank">Bootstrap</a> version 5.1.3. This is available under the <a href="https://github.com/twbs/bootstrap/blob/master/LICENSE" target="_blank">MIT License</a>.
					</li>
					<li>
						Syntax highlighting is implemented via <a href="https://highlightjs.org" target="_blank">highlight.js</a> licensed under the BSD 3-Clause License and the included github styling made by Vasily Polovnyov.
					</li>
					<li>
						Icons on this site are from <a href="https://fontawesome.com/" target="_blank">Font Awesome</a> which is licensed under the <a href="https://fontawesome.com/license/free" target="_blank">Font Awesome Free License</a>.
					</li></ul>
				<p>Built at 2022-10-17 13:22:35 UTC</p>
			</details>
		</footer>

		
	</body>
</html>